### PR TITLE
Added capability for OS-specific settings

### PR DESCRIPTION
**Pull request by ErichDonGubler**
As a cross-platform user of Sublime, I needed Cygwin for Windows, but the default terminal for my Arch Linux install. Currently Terminal doesn't look for cross-platform settings à la vanilla Sublime settings (i.e., `Terminal (Windows).sublime-settings`, `Terminal (OSX).sublime-settings`. This PR adds this capability and allows Terminal to use OS-specific settings.

### CODE REVIEW

**Comment by wbond** [comment]
How about just a function instead of a class with a single static method?

File: Terminal.py [hunk 1]
```diff
@@ -21,16 +21,26 @@ class NotFoundError(Exception):
 else:
     installed_dir = os.path.basename(os.getcwd())
 
+class TerminalSettings():
+    @staticmethod
+    def get(key, default = None):
```
Comment (twolfson) [inline Terminal.py]: I agree with @wbond on this. If we don't have `self` or `cls` in this function, then there is no reason to have it in a class.

**Comment by ErichDonGubler** [comment]
Fixed. Anything else I need to change?

**Comment by twolfson** [comment]
It looks like a permissions change occurred on `PS.bat`. Can you relocate that to another PR?

**Comment by twolfson** [comment]
After considering, I think this PR is superior to #27 and closed #27. Once the permission change is reverted, then we should be good to land this.

**Comment by ErichDonGubler** [comment]
Done. That was easy!

**Comment by twolfson** [comment]
Looks good. I will land this by the end of the weekend.

File: Terminal.py [hunk 2]
```diff
@@ -21,16 +21,25 @@ class NotFoundError(Exception):
 else:
     installed_dir = os.path.basename(os.getcwd())
 
+def getSetting(key, default = None):
```
Comment (twolfson) [inline Terminal.py]: Ah, one thing that sticks out at me is the function isn't snake_case. Python is always snake_case. However, I will patch this while landing as we keep on having these minor back/forths.

File: Terminal.py [hunk 3]
```diff
@@ -21,16 +21,25 @@ class NotFoundError(Exception):
...
+    settings = sublime.load_settings('Terminal.sublime-settings')
+    os_specific_settings = {};
```
Comment (twolfson) [inline Terminal.py]: And there is a semicolon here, another non-Pythonic style =/

**Comment by twolfson** [comment]
While reviewing the PR locally, I realized that we had no way to open these new preferences. As a result, I added a new option to the "Preferences" dropdown for this: "Settings - More -> OS Specific - User" which is inline with Sublime's handling of excess settings:

https://github.com/twolfson/sublime-files/blob/3083.0.0/Packages/Default/Main.sublime-menu#L822-L836

**Comment by twolfson** [comment]
Here's a screenshot:

selection_141

**Comment by ErichDonGubler** [comment]
@twolfson: I have included your suggested changes from the landing into my [`os-specific-refactor`](https://github.com/ErichDonGubler/sublime_terminal/tree/os-specific-refactor) branch because of a refactor -- I saw that the `sublime` interface actually has a [`platform()`](https://www.sublimetext.com/docs/3/api_reference.html) method, which is clearer to use than `os.name`. I can add that commit to this PR, but won't until you have reviewed [the commit](https://github.com/ErichDonGubler/sublime_terminal/commit/f96c3d043a2e808e8c1efbb48081333f8c34d68b).

**Comment by twolfson** [comment]
Yea, that would work too. You will need to relocate the `platform()` and any `sublime` interactions into a method though. ST3 doesn't guarantee those functions work until after `plugin_loaded` and `load_settings` can change over the lifetime of a running application.

**Comment by ErichDonGubler** [comment]
@twolfson: Not true in the case of `sublime.platform()`. I quote from the docs:

> At importing time, plugins may not call any API functions, with the exception of sublime.version(), sublime.platform(), sublime.architecture() and sublime.channel().

**Comment by twolfson** [comment]
Ah, good to know. We will still need to relocate the `load_settings` though =/

**Comment by twolfson** [comment]
Looks like the attached PR with a release message was landed. @ErichDonGubler if you still want to make those new edits to move to `sublime.platform()`, open them in a new PR and I will gladly land them =)

**Comment by twolfson** [comment]
This PR was landed/released in 1.7.0. Thanks again for all your work =)

**Comment by wbond** [comment]
Did I jump the gun here with the release? I saw the release notes on the other PR and figured that meant you were all set. Sorry!

**Comment by twolfson** [comment]
I think it's fine, we were discussing a small refactor to better handle future platforms (and use less code). However, in that event we would prob need another release as is.

**Pull request merged**